datalad_next/annexremote/tests/test_uncurl.py

```diff
@@ -0,0 +1,269 @@
+      custom settings
+    - now move RIA and hence break URL
+    - fix1: only using a URL template, point to dataset dir in RIA store
+      plus some always available key-properties
+    - alternative fix2: simpler template, plus match expression to
+      "understand" some structural aspects of RIA and reuse them
+    """
+    # we create a dataset to bootstrap the test setup, with on file
+    # of known content
+    srcds = EnsureDataset()(tmp_path / 'srcds').ds.create(**res_kwargs)
+    testfile_content = 'mikewashere!'
+    (srcds.pathobj / 'testfile.txt').write_text(testfile_content)
+    srcds.save(**res_kwargs)
+    # pull out some essential properties for the underlying key for later
+    # use in this test
+    testkey_props = srcds.status(
+        'testfile.txt', annex='basic', return_type='item-or-list', **res_kwargs)
+    testkey_props = {
+        k: v for k, v in testkey_props.items()
+        if k in ('key', 'hashdirmixed', 'hashdirlower')
+    }
+
+    # establish a RIA sibling and push
+    baseurl = (tmp_path / "ria").as_uri()
+    srcds.create_sibling_ria(
+        # use a ria+file:// URL for simplicity
+        f'ria+{baseurl}',
+        name='ria',
+        new_store_ok=True,
+        **res_kwargs
+    )
+    srcds.push(to='ria', **res_kwargs)
+    # setup is done
+
+    # start of the actual test
+    # create a fresh dataset
+    ds = EnsureDataset()(tmp_path / 'testds').ds.create(**res_kwargs)
+    dsca = ds.repo.call_annex
+    # we add uncurl WITH NO config whatsoever.
+    # this must be enough to be able to use the built-in downloaders
+    target_fname = 'mydownload.txt'
+    dsca(['initremote', 'myuncurl'] + std_initargs)
+    dsca(['addurl', '--file', target_fname,
+          # we download from the verbatim, hand-crafted URL
+          f'{baseurl}/{srcds.id[:3]}/{srcds.id[3:]}/annex/objects/'
+          f'{testkey_props["hashdirmixed"]}'
+          f'{testkey_props["key"]}/{testkey_props["key"]}'
+    ])
+    assert (ds.pathobj / target_fname).read_text() == testfile_content
+    # make sure the re-downloaded key ends up having the same keyname in
+    # the new dataset
+    assert ds.status(
+        target_fname, annex='basic', return_type='item-or-list',
+        **res_kwargs)['key'] == testkey_props['key']
+
+    # now we drop the key...
+    ds.drop(target_fname, **res_kwargs)
+    assert not (ds.pathobj / target_fname).exists()
+    # ...and we move the RIA store to break the recorded
+    # URL (simulating an infrastructure change)
+    (tmp_path / 'ria').rename(tmp_path / 'ria_moved')
+
+    # verify that no residual magic makes data access possible
+    with pytest.raises(IncompleteResultsError):
+        ds.get(target_fname, **res_kwargs)
+
+    # fix it via an access URL config,
+    # point directly via a hard-coded dataset ID
+    # NOTE: last line is no f-string!
+    url_tmpl = f'{(tmp_path / "ria_moved").as_uri()}/{srcds.id[:3]}/{srcds.id[3:]}' \
```

Comment:
This looks a bit odd with f-string and template string mixture. How about this?

```suggestion
    url_tmpl = (tmp_path / "ria_moved" / srcds.id[:3] / srcds.id[3:]).as_uri() \
```

Reply:
I included a change in that spirit into an update. Thx.